### PR TITLE
Allow turning off classification or detection in GUI

### PR DESCRIPTION
**Pull request by matham**
## Description

**What is this PR**

- [ ] Bug fix
- [x] Addition of a new feature
- [ ] Other

**Why is this PR needed?**

Currently, there's no way to separately run detection and classification. Because classification takes longer than detection, it's a good idea to be able to tune the detection parameters, without having to run classification as well. Similarly, it'd be nice to be able to generate detected points and then tune the classification parameters without having to re-run detection each time as well.

E.g. we need to tune the detection parameters because that step mises a lot of the cells.

**What does this PR do?**

This PR adds two new options to the GUI - to skip detection (and only run the classification part) and to skip classification (and only run the detection part). It also adds an input, in which you can select a `Points` layer as the points input for the classification step when detection is skipped.

Hence a typical process would be, you run just the detection step until the parameters are tuned. Then, use that steps output layer as input and just run the classification step until that's tuned as well.

## References

I developed this PR on top of the https://github.com/brainglobe/cellfinder/pull/401 branch (but committed it on main). But I'll probably need to rebase this PR when that's merged.

## How has this PR been tested?

I tested the process in the GUI.

## Is this a breaking change?

No. All new parameters are optional.

## Does this PR require an update to the documentation?

Not sure if I should update the docs for this as it's optional parameters.

## Checklist:

- [x] The code has been tested locally
- [ ] Tests have been added to cover all new functionality (unit & integration)
- [ ] The documentation has been updated to reflect any changes
- [x] The code has been formatted with [pre-commit](https://pre-commit.com/)

### CODE REVIEW

**Comment by matham** [comment]
I did a bit of re-factoring to fix the sonar cloud code analysis. However, two of the errors haven't gone away. It complains that it may be none, but a) I added an `assert` prior that it is not None and b) if it is None, it would happen during tests, which should be ok.

Not sure what to do because adding the code it wants will make the error go away, but that could would be wasteful.

**Comment by adamltyson** [comment]
Thanks @matham, I think you can ignore SonorCloud at this stage. I interpret it more as advice than anything else. I'd be happy to merge if that's the only think failing.

Anyway, thanks again for contributing, I'm assigning @IgorTatarnikov as reviewer.

File: cellfinder/napari/detect/detect.py [hunk 1]
```diff
@@ -218,21 +375,12 @@ def widget(
             use_pre_trained_weights, trained_model
```
[the remark this answers is not in the document]
Comment (matham) [reply cellfinder/napari/detect/detect.py]: Oof I must have lost that parameter after testing, otherwise I'm not sure how it could have worked 😬

**Review by IgorTatarnikov — APPROVED** [review]
:+1: Thanks for addressing the comments!

**Pull request merged**